### PR TITLE
Add group to output workspace name

### PR DESCRIPTION
**Pull request by dlcaballero16**
## Description of work

This updates the output workspace names in diffcal so that they now include the grouping. From Malcolm's suggestion, I removed "diffoc" from the name since having the grouping in the name implies that it's already focused. 
## Explanation of work

WorkspaceNameGenerator had a function added so that we can add only a grouping and not modify the workspace type. Any calls to diffcal_output() have been updated to include the grouping.

### Dev testing

Run diffcal and make sure that you can see 2 workspaces: dsp_grouping_runNumber and tof_grouping_runNumber. Complete the save step and make sure there's corresponding nxs files and the names in CalibrationRecord.json are correct.

### CIS testing

See Dev Testing

## Link to EWM item

[EWM#3819](https://ornlrse.clm.ibmcloud.com/ccm/web/projects/Neutron%20Data%20Project%20(Change%20Management)#action=com.ibm.team.workitem.viewWorkItem&id=3819)

### CODE REVIEW

File: src/snapred/backend/recipe/algorithm/GroupDiffractionCalibration.py
Comment (dlcaballero16) [inline src/snapred/backend/recipe/algorithm/GroupDiffractionCalibration.py:1]: This should not be included

File: src/snapred/ui/handler/SNAPResponseHandler.py
Comment (dlcaballero16) [inline src/snapred/ui/handler/SNAPResponseHandler.py:1]: This should not be included in this review

File: src/snapred/ui/presenter/WorkflowPresenter.py
Comment (dlcaballero16) [inline src/snapred/ui/presenter/WorkflowPresenter.py:1]: This should not be included in this review

File: src/snapred/ui/workflow/WorkflowImplementer.py
Comment (dlcaballero16) [inline src/snapred/ui/workflow/WorkflowImplementer.py:1]: This should not be included in this review

**Comment by rboston628** [comment]
Ran it, workspaces showed up.  Not sure why everything was turned into lowercase, though.
image

File: src/snapred/backend/data/GroceryService.py [hunk 1]
```diff
@@ -224,8 +224,8 @@ def _createGroupingWorkspaceName(self, groupingScheme: str, runNumber: str, useL
     def _createDiffcalInputWorkspaceName(self, runNumber: str) -> WorkspaceName:
         return wng.diffCalInput().runNumber(runNumber).build()
 
-    def _createDiffcalOutputWorkspaceName(self, runNumber: str, version: str, unit: str) -> WorkspaceName:
-        return wng.diffCalOutput().unit(unit).runNumber(runNumber).version(version).build()
+    def _createDiffcalOutputWorkspaceName(self, runNumber: str, version: str, unit: str, group: str) -> WorkspaceName:
+        return wng.diffCalOutput().unit(unit).runNumber(runNumber).version(version).group(group).build()
```
Comment (rboston628) [inline src/snapred/backend/data/GroceryService.py:228]: try adding
``` python
group = group.split(' ')[0]
```

File: tests/unit/backend/service/test_CalibrationService.py [hunk 2]
```diff
@@ -161,7 +161,9 @@ def create_fake_diffcal_files(self, path: Path, workspaces: Dict[WorkspaceType,
         # Note: this form allows testing of partial-write behavior:
         #   e.g. no _mask_ or _table_ workspace in the 'workspaces' dict.
         for key, wsNames in workspaces.items():
+            print(key)
             for wsName in wsNames:
+                print(wsName)
```
Comment (rboston628) [inline tests/unit/backend/service/test_CalibrationService.py:166]: remove these prints

File: src/snapred/backend/data/GroceryService.py (hunk 1 above)
Comment (rboston628) [reply src/snapred/backend/data/GroceryService.py:228]: Nevermind, don't do that or you'd have to do it in two other places.

**Review by rboston628 — APPROVED** [review]

**Pull request merged**